### PR TITLE
Adds laserguns with rechargeable, swappable magazines

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
They use the old autorifle sprites that Ausops made, since they are nice and future looking and shouldn't be unused.

:cl: Kor
rscadd: Add laserguns with swappable, rechargeable magazines. They are adminspawn only.
/:cl:

### CODE REVIEW

**Comment by Zilenan91** [comment]
screenshots?

File: code/game/machinery/recharger.dm [hunk 1]
```diff
@@ -118,6 +118,14 @@
 					icon_state = "recharger2"
 			else
 				icon_state = "recharger3"
+		if(istype(charging, /obj/item/ammo_box/magazine/recharge))
+			var/obj/item/ammo_box/magazine/recharge/R = charging
+			if(R.stored_ammo.len<R.max_ammo)
```
Comment (neersighted) [inline code/game/machinery/recharger.dm]: spaces please

File: code/game/machinery/recharger.dm [hunk 2]
```diff
@@ -118,6 +118,14 @@
...
 			else
 				icon_state = "recharger3"
+		if(istype(charging, /obj/item/ammo_box/magazine/recharge))
+			var/obj/item/ammo_box/magazine/recharge/R = charging
+			if(R.stored_ammo.len<R.max_ammo)
+				R.stored_ammo += new R.ammo_type(R)
+				icon_state = "recharger1"
+				use_power(200)
+			else
+				icon_state = "recharger2"
```
Comment (neersighted) [inline code/game/machinery/recharger.dm]: Move this stuff out into a update_icon proc?

**Comment by TechnoAlchemisto** [comment]
happening

File: code/game/machinery/recharger.dm (hunk 2 above)
Comment (KorPhaeron) [inline code/game/machinery/recharger.dm]: Rewriting rechargers is beyond the scope of this PR

Or something

**Comment by Bawhoppen** [comment]
:-1: get actual sprites (not that those are bad just not for this)

**Comment by KorPhaeron** [comment]
@neersighted @tkdrg 

I updated the update_icon, dont know if it is better or worse now

**Pull request merged**